mantis-server/mantis-server-agent/src/main/java/io/mantisrx/server/agent/TaskExecutorStarter.java

```diff
@@ -68,6 +70,17 @@ protected void startUp() {
         highAvailabilityServices.startAsync().awaitRunning();
 
         taskExecutor.start();
+        taskExecutor.awaitRunning().exceptionally(ex -> {
```

Comment:
can you instead do
```
taskExecutor.awaitRunning().get();
```
This way, you will bubble up the taskExecutor startup failure to the level of the TaskExecutorStarter's service, which will throw an exception in your AgentV2Main::start.